### ISSUE TITLE
Action for discard all in directory

### ISSUE THREAD

**Issue by TimothyPescatore** [NeogitOrg/neogit#721]
**Is your feature request related to a problem? Please describe.**
When you make changes with a script or any other automated way, it is easy to change/create many files. In case there was a mistake in this script you need to discard many changes. Right now there is only an option to discard a single file or hunk which has also a confirmation window. If you now need to discard changes in more than 5 files it gets quickly laborious. 
What will you do if your script changed 100 files wrongly?

**Describe the solution you'd like**
There is an option to press "x" (default key for discard) on a directory and it will discard all changes within this directory and all sub directories. In lazygit for example the changes are displayed in a tree-like structure, therefore the directory is already displayed. In Neogit the files are just listed as a list. I don't know what if there is a philosophy in the displaying of the changed files but I am sure there are also other possibilities to show clearly which files are under what directory to so you could easily understand what file changes will be discarded. 

**Describe alternatives you've considered**
I just use lazygit for it for now.

**Additional context**
Thank you for this amazing plugin and the enormous speed it is developing!

**Comment by CKolkey** [comment]
You should be able to do a linewise selection of all the files you want to discard (`V`), and discard them all at once - does that work for you?

**Comment by TimothyPescatore** [comment]
If it works then yes. It may take a few more key strokes but at least it doesn't force me to use another tool.
Thank you

**Comment by CKolkey** [comment]
Certainly. The list-view vs. tree-view is pretty core to the different UI's. In git, a directory can't be added on it's own, but files within it can. 

As Neogit's status buffer maps closely to git's underlying data model, there's no "folder" as an item, only filepaths - the exception being an untracked folder with files in it (for now)
